mypy/applytype.py

```diff
@@ -62,3 +66,16 @@ def apply_generic_arguments(callable: CallableType, types: List[Type],
         ret_type=expand_type(callable.ret_type, id_to_type),
         variables=remaining_tvars,
     )
+
+
+def has_anystr_incompatible_args(arg_strings: Sequence[str], type: Type) -> bool:
+    """Determines if function has a problem with AnyStr arguments.
+
+    If the function has more than one AnyStr argument and the solver returns the object type,
+    then the function was passed both an "str" and "bytes" argument type.
+    """
+    if isinstance(type, Instance) and type.type.name() == 'object':
+        for string in arg_strings:
+            if 'AnyStr' in string:
```

Comment:
There are many of them (use ``grep -rn mypy -e "TypeVisitor"`` and/or ``grep -rn mypy -e "TypeQuery"`` to see them). One of the simplest and closest to what you need is probably the one in ``typeanal.get_type_var_names``. Another simple example is ``expandtype.ExpandTypeVisitor``.